src/Analyser/MutatingScope.php

```diff
@@ -1242,7 +1242,7 @@ private function resolveType(string $exprString, Expr $node): Type
 						new VoidType(),
 					]);
 				} else {
-					$returnType = $arrowScope->getKeepVoidType($node->expr);
+					$returnType = $arrowScope->getType($node->expr);
 					if ($node->returnType !== null) {
 						$returnType = TypehintHelper::decideType($this->getFunctionType($node->returnType, false, false), $returnType);
```

Comment:
I came to the conclusion that `getKeepVoidType` might not be necessary because the following IF did not keep void when infered from `$node->returnType`